### ISSUE TITLE
SSH Timeout - Cant connect via ssh no matter what

### ISSUE THREAD

**Issue by dezinerdudes** [Varying-Vagrant-Vagrants/VVV#375]
Dear developers,

Thank you so much for this awesome project that you have created so far. From what Ive read it looks sooo awesome. Thats why I am trying really hard to get it working on my machine, but i simply cannot. I have painstakingly followed your tuturial over and over again step by step for days and tried like a million solutions on google but cannot find out why I experience a certain issue. Please could I ask for someone to give me some more wisdom:

I setup vagrant and VB just as specified. I am able to progress fine in setting up the VM specified in the vvv vagrant file through Git Bash. However, no matter what I try i am not able to ssh into the server. After the terminal comes to 'SSH auth method: private key' it just continues to give me a 'warning: connection timed out. Retrying...' (Please see image below)

I am running windows 8 64bit. I have Git with openssh and keys generated to default .ssh directory. I have tried different terminals and consoles but still not working. My ssh is working though when I connect to other servers. I restarted, deleted, everything the VM and still the same problem. I checked the ports from within the VB GUI settings and even tested with changes port settings but still nothing. Command netstat -a reveals that the vm is open on the corrent ssh ports and listening. Vagrant up command works, but no other command like halt or suspend etc. Reload and status works. Putty with generated vagrant keys according to your wiki article also times out in connection. I have just about read through and tried everything listed in google but cannot get it to work. 

Please please help! I really want to use your setup file!

![image](https://cloud.githubusercontent.com/assets/7773916/3154006/e2d9441e-eaa2-11e3-8937-d2bd1605cb7d.png)

**Comment by ss44** [comment]
Enabling the virtualization settings in the bios worked for me. cheers!

**Comment by xitude** [comment]
I fixed my with a firewall issue. I had to added it to connections.

**Comment by xacaxulu** [comment]
+1

**Comment by mucsi96** [comment]
Enabling the virtualization settings in the bios worked for me as well.(64 bit Windows 8.1) thanks!

**Comment by morganda** [comment]
I just ran into a similar issue.

Setup: I have several boxes in 1 Vagrantfile (testing a replication set in mysql). Each box was pretty much the same except for the name, the IP, and how it's provisioned.

Problems: While spinning up and down some of these boxes, tweaking my provisioners, and the Vagrantfile, I started to get this issue on one of my boxes. I attempted to destroy the the box, vagrant box remove, rm -rf .vagrant/ to no avail. I set the gui flag so I could get on the box and see what wasn't right. The box hadn't taken the IP I had given it! No idea why.

Solution: There was a typo in my Vagrantfile. I'm not exactly sure where the typo was, because I just blew away the config and copied it over from a functioning box and just tweaked it so it had the IP and name that I had before. I wish I had taken the time to identify the typo in my Vagrantfile, but I was annoyed enough that I didn't care at the time.

Hope that helps someone.

**Comment by latentgod** [comment]
Oh...  I also face this problem ,and I spend 4 hours  to solve this problem . First, according to way which appear on internet ,it doesn't work.  At last,I discovery my vmware network card is conflict with  virtualbox network card.Important ,the vboxnet must be in 192.168.10.0/24 and the ip of Homestead.yaml  also must be in 192.168.10.0/24 .So you may be modify vmware network ,and it can't in 192.168.10.0/24.  The above is my solutions.

**Comment by hkic** [comment]
My `private_key` file is missing in `PATH_TO\.vagrant\machines\default\virtualbox`. After restoring it, vagrant box can authenticate the ssh connection.

**Comment by mtrovo** [comment]
@hkic thanks for the comment, my vm had the same problem and I fixed it based on this link http://stackoverflow.com/a/23554973/305116, basically I just added my personal private key to the Vagrantfile and it all worked.

``` ruby
  # SSH Agent Forwarding
  #
  # Enable agent forwarding on vagrant ssh commands. This allows you to use ssh keys
  # on your host machine inside the guest. See the manual for `ssh-add`.
  config.ssh.private_key_path = '~/.ssh/id_rsa'
  config.ssh.forward_agent = true
```

**Comment by fbm-static** [comment]
I'm also having the same problem. I think this is because of the fact that I have installed Homestead prior to trying this setup. I've tried removing the box laravel/homestead and removing the network adapters of virtualbox but have no luck.

**Comment by spigotdesign** [comment]
Thanks @mtrovo adding the `private_key_path` line worked for me too.

**Comment by ronezone** [comment]
@badmadrad Your answer worked for me on Windows 8.1 64-bit!  ENABLED Virtualization Technology in BIOS.  Thank you!

FYI to others when searching for how to update your BIOS: include your machine's manufacturer and model in your google search!  F10 on boot for HP Envy.  Cheers, Ron

**Comment by fbm-static** [comment]
@badmadrad Congralutations on your PhD!

**Comment by poznet** [comment]
Anwser is simple 

stop vagrant
remove insecure_private_key  from  c:\users\username\.vagrant\
vagrant up  

key will be regenerated and setup

**Comment by GuerrillaCoder** [comment]
Hey Guys,

I had same issue but I enabled virtualization in my bios and it fixed it.  Weird thing is that other boxes were working without it enabled.

Check your bios.  It will probably be under advanced -> CPU

**Comment by fbm-static** [comment]
It will work without VT-X enabled but for 64bit boxes it needs VT-X that's why it's failing to start.

**Comment by fbm-static** [comment]
It will work without VT-X enabled but for 64bit boxes it needs VT-X that's why it's failing to start.

**Comment by schikulski** [comment]
I've tried the following:

* Clean install of VirtualBox and Vagrant
* Checked CPU for VT-X support (I'm running a newer Mac so it should be OK).
* Using 32-bit image(just in case)
* Setting the timeout to at least 15min

But I'm still getting the same error as above. 
I tested with a clean ```vagrant init hashicorp/precise32``` - and that worked perfectly:

    ==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2200
    default: SSH username: vagrant
    default: SSH auth method: private key
    ==> default: Machine booted and ready!

But with VVV It's trying to connect to 127.0.0.1:2222. And I'm getting the same error as the other ones:

    ==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...

Anyone have any ideas why this i happening? And why is VVV using port 2222 and the clean vargrantfile using 2200?

**Comment by YordanGeorgiev** [comment]
@dezinerdudes commented on Jul 3, 2014 - your check list was quite comprehensive ... the problem with the https://atlas.hashicorp.com/ubuntu/boxes/trusty64/versions/20150609.0.10 version is that once initialized there is a keyboard input needed ( to select from the bootloader menu !!!! ) OMG I am installing a headless guest image in order to avoid messing with GUI , but no first I have to install VB extension pack , enable display server in VB and after that start remote Desktop Connection to a console to hit down arrow and press Enter for the guest to boot for first time !!! ..... I would consider this a bug with the 20150609.0.10 version ;o) ... Anyway your check list was the most detailed one I found on the broad Internet for this problem ...

**Comment by redconfetti** [comment]
I'm running into this issue myself, though it seems related to the ferm firewall I'm using on my VM. The issue I identified is that because I'm connecting to the VM using 127.0.0.1, instead of the local IP I'm configuring it to use, it's failing.

```
Vagrant.configure('2') do |config|
  config.vm.box = 'ubuntu/trusty64'
  config.ssh.forward_agent = true

  # Required for NFS to work, pick any local IP
  config.vm.network :private_network, ip: '192.168.50.5'
```

I added `config.ssh.host = '192.168.50.5'` to my configuration and this resolved the issue.

I'm not sure if there is some alternative configuration I need to put in place with my firewall to allow connections from 127.0.0.1.

**Comment by plahpoy** [comment]
While this might not be the 'best' way to do it, and it does not 'fix' the key issue, this is what worked for me and got me up and running:

### Vagrant 1.7.4 ###

Edit the Vagrantfile and add the following 3 lines:

config.ssh.username = "vagrant"
config.ssh.password = "vagrant"
config.ssh.insert_key = false

That's what worked for me.

**Comment by gjoh5** [comment]
I was having this issue with an AMD based machine - had to enable "Secure Virtual Machine Mode" in BIOS and it worked without a hitch

**Comment by fjarrett** [comment]
I spent many hours trying to resolve this on OS X El Capitan only to discover that **custom DNS servers** I had set in my Wi-Fi settings were causing SSH to timeout :tired_face: Removed them and everything worked again as normal. Hope this helps someone else avoid the headache I just caused myself...

**Comment by ronezone** [comment]
Thank you @poznet.  Your answer worked for me after my IT addressed issues with my Windows 10 upgrade and broke my glorious VVV env.  After I re-enabled VT-x in BIOS (Virtualization Technology), I was seeing errors indicating ssh issues; so I removed the insecure_private_key from `C:\Users\<username>.<domain>\.vagrant.d\` and it was recreated without issue on next `vagrant up` ...YAY!

Here is output I was seeing:

```
$ vagrant reload
==> default: Removing hosts
==> default: Attempting graceful shutdown of VM...
    default: Guest communication could not be established! This is usually because
    default: SSH is not running, the authentication information was changed,
    default: or some other networking issue. Vagrant will force halt, if
    default: capable.
==> default: Forcing shutdown of VM...
==> default: Checking if box 'ubuntu/trusty64' is up to date...
==> default: A newer version of the box 'ubuntu/trusty64' is available! You currently
==> default: have version '20151203.2.0'. The latest is version '20151217.0.0'. Run
==> default: `vagrant box update` to update.
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 80 => 8080 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying... 
    default: Warning: Connection timeout. Retrying... 
    default: Warning: Connection timeout. Retrying...
   …
    default: Warning: Connection timeout. Retrying...
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
```

```
$ vagrant halt
==> default: Running triggers before halt...
==> default: Executing command "vagrant ssh -c vagrant_halt"...
==> default: ssh_exchange_identification: read: Connection reset by peer
==> default: Command execution finished.
The command "vagrant ssh -c 'vagrant_halt'" returned a failed exit code. The
error output is shown below:

ssh_exchange_identification: read: Connection reset by peer
```

VirtualBox GUI -> Start...
```
Failed to open a session for the virtual machine vagrant-local.
Failed to assign the machine to the session (E_FAIL).
Result Code: 	VBOX_E_VM_ERROR (0x80BB0003)
Component: 	Machine
Interface: 	IMachine {480cf695-2d8d-4256-9c7c-cce4184fa048}
Callee RC: 	E_FAIL (0x80004005)
```

```
Failed to open a session for the virtual machine vagrant-local.
The VM session was aborted.
Result Code: 	E_FAIL (0x80004005)
Component: 	SessionMachine
Interface: 	ISession {12f4dcdb-12b2-4ec1-b7cd-ddd9f6c5bf4d}
```

**Comment by llawrenc** [comment]
Hi All, for what it is worth, I had this same issue with SSH remote disconnect error on boot and solved it for my circumstances.  I fixed the bios setting for virtualization and tried all other solutions above to no avail.  I destroyed and reprovisioned the vagrant machine from scratch and this solved it, no intervention required - it's even using the default Vagrantfile.

LL

**Comment by jartaud** [comment]
Thanks @ronezone removing the key works for me.

**Comment by is-aka-konor** [comment]
Thanks @lkwdwrd,
I turned Hyper V off and it helped me.

**Comment by JBWebServices** [comment]
I am having a lot of issues SSH into my vagrant ubuntu box.. but just to eliminate possibilities I initialized a lucid32 and was able to SSH into that without any problems.

**Comment by vekexasia** [comment]
Here i can't connect from an ubuntu host. Virtualbox is 5.0.10
ping gives `From 192.168.50.1 icmp_seq=1 Destination Host Unreachable`

In virtualbox gui the machine is up and running while not showing anything in the "preview" 

I waited more than 20 mins for the vm to boot i believe there's something not working properly.

**Comment by unfulvio** [comment]
I experience this problem when my machine is suspended

At first I thought it'd happen when my machine get suspended and VVV was already up, then at host resume it's not accessible

However I noticed that even if I haven't yet started VVV and I suspend the host, SSH will have problems when I resume and can't vagrant up

The issue happens with Ubuntu 15.10 and VMWare Workstation -- had similar problems with Virtualbox 5 as well

**Comment by mohamdio** [comment]
I solved the problem by changing the version of ubuntu in vagrantfile to trusty32 instead of 64 in VM Virtualbox because my laptop (old) can't virtualize 64 bits like what @fxbenard did, after down ot 32 get the same problem again and solved it by doing this solution from here 
https://github.com/mitchellh/vagrant/issues/3860#issuecomment-167664778
Add or change in the vagrantfile provider section:
```
config.vm.provider "virtualbox" do |vb|
    ### Change network card to PCnet-FAST III
    # For NAT adapter
    vb.customize ["modifyvm", :id, "--nictype1", "Am79C973"]
    # For host-only adapter
    vb.customize ["modifyvm", :id, "--nictype2", "Am79C973"]
  end
```
i hope this will help someone

**Comment by balmas** [comment]
@mohamdio thanks so much this did the trick for me too!

**Comment by cfoellmann** [comment]
this should be written up for the docs!?

**Comment by onshop** [comment]
Oddly, to get vagrant up progressing further, in the BIOS settings I had to:

Disable: Virtualisation
Enable: VT-X

**Comment by cristovaov** [comment]
@onshop I believe this is something that VirtualBox menions in its documentation

**Comment by jeffreygnatek** [comment]
i was having this issue and i followed the instructions to completely remove vagrant, including plugins etc, and remove virtualbox, then brew installed them and installed the trigger and host-updater plugin, then everything appeared to work ok.

**Comment by malanve** [comment]
Make sure the Virtual Machine's sshd service starts up on bootup

**Comment by ocombe** [comment]
OMG thank you @shivapoudel, I disabled Hyper-V and vagrant is able to connect to ssh again! I think this option was activated by visual studio to run the Windows Phone VMs. If anyone has this problem, here is how to disable it: http://www.eightforums.com/tutorials/42041-hyper-v-enable-disable-windows-8-a.html then just reboot and it will work again !

**Comment by johnjamesjacoby** [comment]
I'll [drop this here](https://jaco.by/2016/05/05/slow-vagrant/). For me, it was the sheer number of directories in VVV's `www` directory (about 339k.)

**Comment by giovannipds** [comment]
Hello peoople! I didn't read it all but, I just came by to gave you my little contribution. Based on your comments, I was able to diagnostic what my problem was (the same issue / related solutions). And this is the knowledge that helped me through that. First, I had 2 different problems:

1. `vagrant up` wasn't able to find my ssh '**id_rsa**' (because I didn't have it yet, at that time):
I ran `ssh-keygen -t rsa -b 4096 -C "myemailaddress@mydomain.com"`, based on [this GitHub's article](https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/), and voilá, steped through that;

2. Then, I got the same problem as you guys "**Warning: Connection timed out. Retrying...**", eternally...:
So, reading your comments, I had to restart my system and look at my BIOS (F2 to get there, on PC), and there I've noticed that **Virtualization was disabled**. I've enabled that, saved, and started the system once again, to check if has changed anything.

After that, `vagrant up` worked like a charm! It's 4am but it is running! How cool is that, hã? :D As I know there are very few masochist developers like me, that would try this at Windows, specially at **Windows 10**, I just couldn't not to forget coming here and left my word... another important information, is that, I was trying to set-up **Laravel 5**, using Homestead, VirtualBox, composer etc. It has worked. So, hope this comment helps someone like the comments on this issue helped me. My best wishes. G-bye!

**Comment by tonygilkerson** [comment]
I was having the same issue on windows and turning off my local firewall as suggested above fix the problem.  Thanks very much.

**Comment by sedulousfunds** [comment]
I spent two days trying to get vagrant ssh or any ssh client to connect with vagrant/VM on Windows 10 with no luck - both git ssh and vagrant ssh connections were refused. After working through every recommendation on this thread, I found out the problem was that I had to completely remove my anti-virus (Iolo System Mechanic/System Shield). Note that simply disabling real-time protection made no difference. Once I deleted System Mechanic and replaced with Windows Defender (included in Windows 10) as my anti-virus app, vagrant up completely provisioned vvv and I was good to go - I will leave it to smarter people to explain why. Hope this helps!

**Comment by WRRB** [comment]
I'm on windows 7. What worked for me was changing the box.
boxcutter/ubuntu1404-desktop didn't work (I couldn't ssh into that one no matter what I tried)
janihur/ubuntu-1404-desktop & boxcutter/centos72-desktop both did work from the get go.

**Comment by stuartjnelson** [comment]
I just started getting this error last week. I've trawled the internet for fixes but as of yet nothing has worked. I'm running;

- OSX 10.11.6
- Vagrant 1.8.5
- VirtualBox 5.0.16
- Ansible 2.0.2.0 


Has anyone else come across this issue recently? I've tried everything I can find form creating a new user to PRAM restart to trying to bypass SSH to manual settings the username & password in my vagrant file;

`config.ssh.username = "vagrant"
config.ssh.password = "vagrant`


Any advice anyone has would be much appreciated!

**Comment by msdos** [comment]
@stuartjnelson Unfortunately, I had to run `vagrant ssh-config` and delete the key that it's shown, then run `vagrant destroy` to completely solve my problem.

**Comment by rmansnacks** [comment]
I solved it, it's really trivial. Go to your virtualbox settings of machine -> network -> under your first adapter (set to nat) click on advanced end check "Cable connected" and click OK. 

It's sounds easy, but it took me while to troubleshoot. I've installed new vagrant along with new vbox, I've set new machine in my no firewall no virus scanner environment. I've tried to connect via vagrant with no result. Then i've decided i'll copy my vagrant ssh key to other machine. I've couldn't connect (i had no adapter set on vb). So i made public adapter on my vbox (this is when i realised that default option is set to cable not connected). So i went to my machine to check the state of my network adapter, it was fun to realize that it was disconnected :)  It came with last Virtualbox update. Im most certainly sure that my solution will solve problem for most of you. It's bad habit of us not to check vbox settings.

**Comment by AlanTondelier** [comment]
Thanks rmansnacks !
I was looking for the same issue for 2 hours now, but checking "Cable Connected" fixed it for me. 

Strange that it's "unplugged" by default...

**Comment by JJJ** [comment]
Everything is better with screenshots.

<img width="778" alt="screen shot 2016-11-02 at 11 35 54" src="https://cloud.githubusercontent.com/assets/88951/19937783/99b0a2d0-a0f0-11e6-9efc-340f9680d72a.png">

**Comment by rmansnacks** [comment]
It's cross platform issue, should i to turn on 3 different computers to make those screens? We are people of it not screen readers ;)

**Comment by stephenrust** [comment]
Confirmed @AlEltono's solution worked for me.

**Comment by timgvandijk** [comment]
Wauw, worked through at least a dozen Stack Overflow / Laravel.io / Laracasts threads, reinstalled vagrant, virtualbox, and homestead, and tried tons of proposed solutions, only to find out the source of all this madness is one lonely unchecked checkmark ^^

Thanks @rmansnacks, I can sleep in peace now.

**Comment by v1ncen** [comment]
Same here. Thanks too @rmansnacks! Worked for me on macOS Sierra with VirtualBox 5.0.30 r112061

**Comment by stevenbuehner** [comment]
Installing the newest VirtualBox 5.0.32-112930-OSX.dmg solved the issue entirely for me on OSX 10.12.3 ...

**Comment by lvillarino** [comment]
Confirmed @AlEltono's solution worked for me on El Capitan 10.11.6

**Comment by pankaj9492** [comment]
Hi All,
I did try all things which is mention above threads, but problem is not solved so  any can give me correct solution for it.

I am using cantos 6.2 and with virtual box 5.1 and Vagrant 1.9.3 for installing PredictionIO.

vagrant up has stuck and timeout error after increasing time too.

**Comment by ralphxlx** [comment]
Hi @pankaj9492 , I got same problem. Did you fixed it? But i has been installed homestead-2.0.box

**Comment by tomjn** [comment]
I'm closing this out as most of the issues are with older versions of VB/Vagrant/VVV, and need to be retested with the latest versions
